start_db.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+set -e
+
+POSTGRES_OPTIONS="-c config_file=/etc/postgresql/postgresql.conf"
+
+# Externally mounted volumes are owned by root, make sure the data directory
+# is owned by postgres user inside the container
+echo "\n===== Modifying ownership/permissions of data directories ====="
+chown -R postgres "/var/lib/postgresql"
+
+# Check if this is the first time running
+if [ ! -s "$PGDATA/PG_VERSION" ]; then
+
+    echo "\n============= Initializing Postgres Data Directory ============"
+    gosu postgres initdb
+
+    echo "\n=============== Starting Postgres in Background ==============="
+    gosu postgres postgres $POSTGRES_OPTIONS &
+    POSTGRES_PID="$!"
+    sleep 5  # wait for postgres to start up before using psql
```

Comment:
I haven't looked into it deeply, but `pg_ctl` may make some of these issues easier to deal with. http://www.postgresql.org/docs/9.4/static/app-pg-ctl.html